tools/bcr_validation.py

```diff
@@ -203,7 +203,7 @@ def add_module_dot_bazel_patch(self, diff, module_name, version):
     patches = source.get("patches", {})
     patches["module_dot_bazel.patch"] = integrity(read(patch_file))
     source["patches"] = patches
-    with open(self.registry.get_source_path(module_name, version), "w") as f:
+    with open(self.registry.get_source_json_path(module_name, version), "w") as f:
       json.dump(source, f, indent=4)
       f.write("\n")
```

Comment:
```suggestion
    self.registry.get_source_json_path(module_name, version).write_tex(json.dumps(source, f, indent=4) + "\n")
```

Go pathlib!

Reply:
Haha totally agree! I actually tried to hold back and not change all the things

Reply:
You could change what you touch